canonical-web-and-design/maas-ui
# feat(ui): add ability to change a machine's storage layout

## Done

- Added "Change storage layout" dropdown and confirmation form
- Converted FormCard to TS

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the storage tab of a machine that is in Ready or Allocated state and check that there is a "Change storage layout" dropdown
- Check that changing to each storage layout updates the storage data correctly
- Go to the storage tab of a machine that is not Ready or Allocated, and check that there is no dropdown

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2160
